op-service/sources/types.go

```diff
@@ -158,34 +231,22 @@ func (block *RPCBlock) Verify() error {
 	if computed := types.DeriveSha(types.Transactions(block.Transactions), trie.NewStackTrie(nil)); block.TxHash != computed {
 		return fmt.Errorf("failed to verify transactions list: computed %s but RPC said %s", computed, block.TxHash)
 	}
-	if block.WithdrawalsRoot != nil {
-		if block.Withdrawals == nil {
-			return errors.New("expected withdrawals")
-		}
-		for i, w := range *block.Withdrawals {
-			if w == nil {
-				return fmt.Errorf("block withdrawal %d is null", i)
-			}
-		}
-		if computed := types.DeriveSha(*block.Withdrawals, trie.NewStackTrie(nil)); *block.WithdrawalsRoot != computed {
-			return fmt.Errorf("failed to verify withdrawals list: computed %s but RPC said %s", computed, block.WithdrawalsRoot)
-		}
-	} else {
-		if block.Withdrawals != nil {
-			return fmt.Errorf("expected no withdrawals due to missing withdrawals-root, but got %d", len(*block.Withdrawals))
-		}
+
+	if err := checker.ValidateWithdrawals(block.Withdrawals, block.WithdrawalsRoot); err != nil {
+		return err
 	}
+
 	return nil
 }
 
```

Comment:
Perhaps a better way to do the check is an L2 block will always have >1 transaction and we could check the type of the first transaction to see if its a deposit. Then we don't need this extra argument.

Another way would be to have different implementations of the `Verify` function on two different types